sdk/eventhub/Azure.Messaging.EventHubs/src/Producer/EventDataBatch.cs

```diff
@@ -130,19 +130,14 @@ public bool TryAdd(EventData eventData)
             {
                 AssertNotLocked();
 
-                bool instrumented = EventDataInstrumentation.InstrumentEvent(eventData, FullyQualifiedNamespace, EventHubName);
-                bool added = InnerBatch.TryAdd(eventData);
+                eventData = eventData.Clone();
```

Comment:
In Java, we don't clone the Event before adding it to the EventDataBatch. I wonder If there is a case where people would add it to a batch, hold the reference and want to modify the event.

Reply:
But it makes sense since we calculate the batch size when it first gets added.... so if more contents get added to that EventData reference, its possible to get over the batch size.

Reply:
Track one didn't clone either, but that leaves a gap for adding something and then messing with the user properties bag after, which invalidates the size measurements and corrupts the batch.   It came up in Service Bus discussions at one point and when I looked, we never did anything defensive when swapping from storing the AMQP message to keeping the raw event.    

Personally, I hate this, as we're now paying for double the allocations when adding the event, but it does prevent the issue.   #9569 will make this obsolete, but it hasn't bubbled up in the priorities to be worked on yet.

Reply:
I finally found the correct [file](https://github.com/dotnet/runtime/blob/master/src/libraries/System.Private.CoreLib/src/System/Array.cs) in the .NET Core source as well, and the behavior still appears to allocate on each [call](https://github.com/dotnet/runtime/blob/master/src/libraries/System.Private.CoreLib/src/System/Array.cs#L676)